src/mixins/yield-erc/ERC20Vault.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.10;
+
+import {ERC20} from "../../tokens/ERC20.sol";
+import {FixedPointMathLib} from "../../utils/FixedPointMathLib.sol";
+import {SafeTransferLib} from "../../utils/SafeTransferLib.sol";
+
+/// @title Yield Bearing Vault
+/// @author joeysantoro, Transmissions11 and JetJadeja
+contract ERC4626Vault is ERC20 {
+    using SafeTransferLib for ERC20;
+    using FixedPointMathLib for uint256;
+
+    /*///////////////////////////////////////////////////////////////
+                                IMMUTABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The underlying token the Vault accepts.
+    ERC20 public immutable underlying;
+
+    /// @notice The base unit of the underlying token and hence vault.
+    /// @dev Equal to 10 ** decimals. Used for fixed point arithmetic.
+    uint256 public immutable baseUnit;
+
+    /// @notice Creates a new Vault that accepts a specific underlying token.
+    /// @param _underlying The ERC20 compliant token the Vault should accept.
+    /// @param _name The name for the vault token.
+    /// @param _symbol The symbol for the vault token.
+
+    constructor(
+        ERC20 _underlying,
+        string memory _name,
+        string memory _symbol
+    )
+        ERC20(
+            _name,
+            _symbol,
+            _underlying.decimals()
```

Comment:
why is it necessary to have `baseUnit`? `decimals` is not part of the ERC20 standard.
From personal experience I can say that at least in the years of 2018-2019, some dexes were unable to list tokens because they had `DECIMALS` instead of `decimals` or no decimals at all.

But the real issue is that from what I see there is no real need for base unit here, and it would be better to just have shares and exchange rate denominated with fixed 18 decimals.